src/ConfirmProvider/ConfirmProvider.stories.tsx

```diff
@@ -15,35 +17,147 @@ const meta: Meta<typeof ConfirmProvider> = {
+    title: "Dialog Title"
+  }
+};
+
+// story with static method
+export const StaticMethod = {
+  render: options => {
+    return (
+      <ConfirmProvider>
+        <Button
+          onClick={() => staticConfirm(options).then(confirmationAction)}
+          variant="contained"
+        >
+          Click
+        </Button>
+      </ConfirmProvider>
+    );
+  }
+};
+
+// story with custom text
+export const WithCustomText = Template.bind({});
+WithCustomText.args = {
+  cancellationText: "No",
+  confirmationText: "Yes",
+  description: "This is a custom description",
+  title: "Custom Title"
+};
 
-// template for story
-const Template: StoryFn<ConfirmProviderProps> = ({ children }) => {
-  return <ConfirmProvider>{children}</ConfirmProvider>;
+// story with confirmation button props
+export const WithConfirmationButtonProps = Template.bind({});
+WithConfirmationButtonProps.args = {
+  confirmationButtonProps: { color: "success" }
 };
 
-export const Default = {
-  args: {
-    children: <ConfirmDialog />
-  },
+// story with custom Elements
+export const WithCustomElements = Template.bind({});
+WithCustomElements.args = {
+  description: <LinearProgress />
+};
+
+// story with custom content
+export const WithCustomContent = Template.bind({});
+WithCustomContent.args = {
+  content: (
+    <div>
+      <LinearProgress />
+      <Box p={2}>This isn't wrapped in DialogContentText.</Box>
+    </div>
+  )
+};
+
+// function to demonstrate unmounting of the parent component
+function ParentUnmountComponent(options) {
+  const [flip, setFlip] = React.useState(false);
+
+  // state to control the dialog is open or not
+  const [dialogOpen, setDialogOpen] = React.useState(true);
+
+  // toggle the flip state every 2 seconds
+  React.useEffect(() => {
+    const interval = setInterval(() => setFlip(flip => !flip), 2000);
+    return () => {
+      clearInterval(interval);
+    };
+  });
+
+  // handlers for onConfirm and onCancel
+  const handleConfirm = () => {
+    console.log("Confirmed");
+    setDialogOpen(false); // Close the dialog
+  };
+
+  const handleCancel = () => {
+    console.log("Cancelled");
+    setDialogOpen(false); // Close the dialog
+  };
+
+  return (
+    <Fragment>
+      {flip ? (
+        <ConfirmationDialog
+          open={dialogOpen}
+          onConfirm={handleConfirm}
+          onCancel={handleCancel}
+          options={{
+            description: "This is the description for Dialog A.",
+            title: "Dialog A"
+          }}
+        />
+      ) : (
+        <ConfirmationDialog
+          open={dialogOpen}
+          onConfirm={handleConfirm}
+          onCancel={handleCancel}
+          options={{
+            description: "This is the description for Dialog B.",
+            title: "Dialog B"
+          }}
+        />
+      )}
+    </Fragment>
+  );
+}
 
-  render: Template
+export const WithParentUnmount = {
+  render: () => <ParentUnmountComponent />
 };
```

Comment:
Suggest that we remove this story. We don't really care for this functionality and its breaking the docs page because the dialogs are always open.